arduino/builder/sketch.go

```diff
@@ -79,14 +117,28 @@ func SketchLoad(sketchPath, buildPath string) (*sketch.Sketch, error) {
 			return nil, errors.Wrap(err, "unable to find the main sketch file")
 		}
 		f.Close()
+		// ensure it is not a directory
+		info, err := os.Stat(mainSketchFile)
+		if err != nil {
+			return nil, errors.Wrap(err, "unable to check the main sketch file")
+		}
+		if info.IsDir() {
+			return nil, errors.Wrap(errors.New(mainSketchFile), "sketch must not be a directory")
+		}
 	} else {
 		sketchFolder = filepath.Dir(sketchPath)
 		mainSketchFile = sketchPath
 	}
 
 	// collect all the sketch files
 	var files []string
-	err = filepath.Walk(sketchFolder, func(path string, info os.FileInfo, err error) error {
+	err = simpleLocalWalk(sketchFolder, maxFileSystemDepth, func(path string, info os.FileInfo, err error) error {
+
+		if err != nil {
+			feedback.Printf("\nerror: %+v\n\n", err)
```

Comment:
```suggestion
			feedback.Errorf("Error during sketch processing: %v", err)
			os.Exit(errorcodes.ErrGeneric)
```

Can you please add the error message as presented in the comment, and force a CLI exit with error?  In case of a symlink loop or other error during the sketch porcessing, the CLI must exit presenting an error.